File: 242.Valid Anagram/solution.java
class Solution {
    public boolean isAnagram(String s, String t) {
        int a[] = new int[26];
        if(s.length()!=t.length()){
            return false;
        }

        int n = s.length();
        for(int i=0;i<n;i++){
            int ind = (int)s.charAt(i) - (int)'a';
            int indx = (int)t.charAt(i) - (int)'a';
            a[indx]--;
            a[ind]++;
        }
        // for(int i=0;i<n;i++){
            // int ind = (int)t.charAt(i) - (int)'a';
            // a[ind]--;
        // }

        for(int i:a){
            if(i!=0){
                return false;
            }
        }
        return true;
        
    }
}

File: 242.Valid Anagram/solution.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if(len(s) != len(t)):
            return False

        temp = {}
        for i in range(len(s)):
            temp[s[i]] = temp.get(s[i],0) +1
            temp[t[i]] = temp.get(t[i],0) -1

        for i in temp.values():
            if i!=0:
                return False
        return True
            

    